Article.md

```diff
@@ -0,0 +1,178 @@
+
+## Case study: A counter managing permissions with ACL
+
+Let’s look at a case study to see how `near-plugins` can be useful to smart contract developers. We are building a `Counter` which stores its current value and has methods to increment, decrement and reset the value. It is intentionally kept simple to allow us to focus on how `near-plugins` adds functionality. This is what the contract looks like prior to using any plugins:
+
+```rust
+#[near_bindgen]
+impl Counter {
+    #[init]
+    pub fn new() -> Self {
+        Self { value: 0 }
+    }
+
+    /// Anyone can retrieve the current value.
+    pub fn value(&self) -> i64 {
+        self.value
+    }
+
+    /// Increases the value of the counter by one.
+    pub fn increment(&mut self) {
+        self.value += 1;
+    }
+
+    /// Decreases the value of the counter by one.
+    pub fn decrement(&mut self) {
+        self.value -= 1;
+    }
+
+    /// Resets the value of the counter to zero.
+    pub fn reset(&mut self) {
+        self.value = 0;
+    }
+}
+```
+
+*The final version of the code is available [in this repository on github](https://github.com/mooori/counter-acl-example). The `Counter` example is inspired by [near-examples/counter-rust](https://github.com/near-examples/counter-rust).*
+
+### Permissions
+
+The contract methods defined above can be called by anyone since they are public and inside an implementation block marked with `#[near_bindgen]`. Using `near-sdk-rs` it is possible to restrict some methods such that they can be called only by the contract itself. Either by using `#[private]` or by not exposing the method publicly, as described in the [documentation](https://docs.near.org/sdk/rust/contract-interface/private-methods).
+
+However, what if some accounts other than the contract should be able to call restricted methods? This is were [`AccessControllable`](https://github.com/aurora-is-near/near-plugins#accesscontrollable) plugin comes in handy.
+
+### Managing permissions with ACL
+
+ACL stands for *access control lists* and they are used in the following way within the `AccessControllable` plugin. The user defines the roles required for their use case as Rust enum variants. Then it is possible to restrict access to a method to accounts that have been granted roles. Restricting access is possible with one line of code, for example:
+
+```rust
+#[access_control_any(roles(Role::Decrementer))]
+pub fn decrement(&mut self) {
+    // ...
+}
+```
+ 
+Let’s walk through it step by step to see how you can make your NEAR smart contract `AccessControllable`.
+
+### Stop 1: Add `near-plugins` as a dependency
+
+For now, `near-plugins` has not yet been published on crates.io. Still, the crate is ready for usage and it can be added as git dependency:
+
+```toml
+# Add `near-plugins` under `dependencies` in your Cargo.toml.
+
+[dependencies]
+near-plugins = { git = "https://github.com/aurora-is-near/near-plugins.git", rev = "102d95b" }
```

Comment:
I agree that this would be resolved when the crate is released and becomes a “regular” dependency. Unless we manually move the tag  as `near-plugins` is changed, I think it would basically be the same: a reference to a commit.

Still, using a tag to the latest release helps explaining why that particular commit is used. It’s about time to release a new version of `near-plugins`. I’ll look into that and then replace the `rev` here with a tag to the new version.

Reply:
The [new release](https://github.com/aurora-is-near/near-plugins/releases/tag/v0.2.0) of `near-plugins` has been published. This repo now specifies the `near-plugins` dependency with `tag = "v0.2.0"` (cc1c75e).